From 23f2198663f453a15c3b7c523ea747cc40c7eb9b Mon Sep 17 00:00:00 2001
From: Jacob Young <jacobly0@users.noreply.github.com>
Date: Sat, 11 Oct 2025 19:19:51 -0400
Subject: [PATCH 026/133] x86_64: fix encoding for out with an immediate port
X-Developer-Signature: v=1; a=openpgp-sha256; l=1533; i=xtex@astrafall.org;
 h=from:subject; bh=GxMkTHoLFBhES5j9I5B7JI+K4PIuRxj2lvyhb3GXQLA=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn7cv985AppOFkeubLAxvHXQYdQ3Xlr87n3lG2xitvhy
 Mo4/b16x0QWBjEuBksxRZYiwwZvVp10ftFl5bIwc1iZQIZIizQwMDAwsDDw5SbmlRrpGOmZahvq
 GRrqGOkYMXBxCsBUSwswMlwLtbhftzawNn7pt+XMVf7LW8smB+9PFcvaf/587vmt718x/FPY43U
 uvnYjS3gU74J1BmJ6/Csixa4lFnFLrbp+f8veH3wA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Closes #25547

(cherry picked from commit ada9035af5cc80f9e8df9818a6dbc4cac5ffd287)
---
 src/arch/x86_64/encodings.zon | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/arch/x86_64/encodings.zon b/src/arch/x86_64/encodings.zon
index ea136569ce93..1a27ca9826a6 100644
--- a/src/arch/x86_64/encodings.zon
+++ b/src/arch/x86_64/encodings.zon
@@ -593,12 +593,12 @@
     .{ .@"or", .rm, .{ .r32,  .rm32   }, .{ 0x0b }, 0, .none,  .none },
     .{ .@"or", .rm, .{ .r64,  .rm64   }, .{ 0x0b }, 0, .long,  .none },
 
-    .{ .out, .zi, .{ .imm8, .al  }, .{ 0xe6 }, 0, .none,  .none },
-    .{ .out, .zi, .{ .imm8, .ax  }, .{ 0xe7 }, 0, .short, .none },
-    .{ .out, .zi, .{ .imm8, .eax }, .{ 0xe7 }, 0, .none,  .none },
-    .{ .out, .z,  .{ .dx,   .al  }, .{ 0xee }, 0, .none,  .none },
-    .{ .out, .z,  .{ .dx,   .ax  }, .{ 0xef }, 0, .short, .none },
-    .{ .out, .z,  .{ .dx,   .eax }, .{ 0xef }, 0, .none,  .none },
+    .{ .out, .i, .{ .imm8, .al  }, .{ 0xe6 }, 0, .none,  .none },
+    .{ .out, .i, .{ .imm8, .ax  }, .{ 0xe7 }, 0, .short, .none },
+    .{ .out, .i, .{ .imm8, .eax }, .{ 0xe7 }, 0, .none,  .none },
+    .{ .out, .z, .{ .dx,   .al  }, .{ 0xee }, 0, .none,  .none },
+    .{ .out, .z, .{ .dx,   .ax  }, .{ 0xef }, 0, .short, .none },
+    .{ .out, .z, .{ .dx,   .eax }, .{ 0xef }, 0, .none,  .none },
 
     .{ .outs,  .z, .{ .dx, .m8  }, .{ 0x6e }, 0, .none,  .none },
     .{ .outs,  .z, .{ .dx, .m16 }, .{ 0x6f }, 0, .short, .none },
-- 
2.52.0

